docs/tutorial/AdvancedTopics.md

```diff
@@ -0,0 +1,117 @@
+   let value = getState().some.random.value;
+   dispatch(someRandomAction());
+}
+```
+
+Cmd.getState and Cmd.dispatch are just symbols that the enhanced store will
+replace with the actual functions at the time the cmd is executed. You can 
+use these symbols as parameters to your cmds as you would any other parameter (the
+order does not matter).
+
+This minimal example will cause perpetual dispatching! While it is also possible
+to make this mistake with large, complicated networks of `redux-thunk` action
+creators, it is much easier to spot the mistake before it is made. It helps to
+keep your reducers small and focused, and use `combineReducers` or manually
+compose reducers so that the number of actions you deal with at one time is
+small. A small set of actions which initiate a `loop` will help reduce the
+likelihood of causing circular dispatches.
+
+## combineReducers with redux-loop
+
+The `combineReducers` implementation in `redux-loop` is aware that some of
+your reducers might return cmds, and it knows how to properly compose them
+and forward them to the store.
+
+```js
+import { createStore } from 'redux';
+import { combineReducers, install } from 'redux-loop';
+
+import { firstReducer, secondReducer } from './reducers';
+
+const reducer = combineReducers({
+  first: firstReducer,
+  second: secondReducer,
+});
+
+const store = createStore(reducer, initialState, install());
+```
+
+## Nest reducers inside parents with their own functionality
+
+`combineReducers` is great for the case where your parent reducer has no functionality of its own, just children.
+However, it's often the case that you want to nest a specific reducer inside a generic parent reducer.
+
+`mergeChildReducers` lets you do this. It merges child reducer state into the parent state while
+composing any Cmds returned by any of the reducers and passing them along to the store.
+
+```js
+
+const initialState = {foo: null};
+
+function parentReducer(state = initialState, action){
+  //typical reducer
+}
+
+export default function reducer(state, action){
+   let parentResult = parentReducer(state, action);
+   return mergeChildReducers(parentResult, action, {child: childReducer});
+}
+
+//final state shape is {foo, child}; foo.child is state returned by childReducer
+```
+
+`combineReducers` just returns a reducer with a specific usage of `mergeChildReducers` (one in which
+the child map never changes, the parent state is an empty object, and all children always get called). 
+By using `mergeChildReducers` in your reducer, you can choose which children are used when and under what key names.
+
+By using `combineReducers` or `mergeChildReducers` everywhere your reducer tree nests, all Cmds resulting 
+from a single dispatched action will always be batched up into a single Cmd at the top of the tree and 
+processed correctly by the store. 
+
+## Using `combineReducers` and `mergeChildReducers` with ImmutableJS and other libraries
```

Comment:
What ar eyour thoguhts on moving the majority of the detail here to the api page about combine reducers and then making a small mention of it on the tutorial page instead of here (and a link to the detailed page). I think so many people will use immutable that it's worth adding to the tutorial so they don't have to go looking for it.

Reply:
yes, I think that's smart. Will update.